model/job-management/src/main/proto/beam_job_api.proto

```diff
@@ -153,6 +154,7 @@ message GetJobStateRequest {
 
 message GetJobStateResponse {
```

Comment:
I think we will want a different version of GetJobStateRequest and GetJobStateResponse for the streaming and non-streaming versions of the state APIs.

Reply:
Why?  What would be different about them?

Reply:
ok, saw your comment from Jira:

> Typically when you would connect you would want to give some lower bound to filter messages. For example GetJobStateRequest could become:
> 
> ```proto
> message GetJobStateRequest {
>   string job_id = 1; // (required)
>   // (Optional) If specified, only state transitions after
>   // the provided timestamp will be provided.
>   google.protobuf.Timestamp min_message_timestamp;
> }
> ```

I agree with this conceptually, but was hoping to defer the addition of this until I get to the message stream.  The state stream is typically going to be just a few events, so practically speaking, the additional of `min_message_timestamp` here will not be very impactful.  The size of the backfill for the message stream on the other hand, with the changes that I'd like to add to it, could be significant, and there could be a number of ways that a user may want to filter (`min_message_timestamp`, `max_num_messages`, `importance`, etc).  I'd like the APIs to be similar between the two, and I think the GetMessageStream will inform what we want to do for GetStateStream. 

tl;dr I'd like to defer the addition of this until later, if possible.